### PR TITLE
#526: Added unit test for HasQueryAggergatorBlock

### LINKED ISSUE #526: [STORY] Add unit tests for HasQueryAggergatorBlock
# Description
`HasQueryAggergatorBlock` is defined in segstructs.go. Add thorough unit testing for this function in segstructs_tests.go in the same folder

Note: You need to name your function(s) so they start with `Test_`; that way Go's unit testing procedure will find that function when it looks for test functions

# Acceptance Criteria
* The added unit tests are clear and thoroughly test the function
* Your new unit tests run and pass when you run `make test` or `make pr` from the siglens base directory
* If adding comments to your code makes it easier to understand, you should add them. If the code is already self-explanatory, you can skip them
* Running `make pr` gives no errors

### PR DESCRIPTION
**Pull request by chilah**
# Description
Fixes #526 Added unit test for HasQueryAggergatorBlock 

# Testing
Added test

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/structs/segstructs_test.go [hunk 1]
```diff
@@ -37,3 +37,23 @@ func Test_HasTransactionArguments_NonNilTransactionArguments(t *testing.T) {
...
+		HarcodedCol: []string{"test1", "test2"},
+		MaxRows:     2,
+		RowsAdded:   1,
+	}
+
+	qa := &QueryAggregators{
+		OutputTransforms: ot,
+	}
+
+	assert.Equal(t, true, qa.HasQueryAggergatorBlock(), "Expected true when QueryAggergatorBlock is not nil, got false")
```
Comment (AndrewHess) [inline pkg/segment/structs/segstructs_test.go:58]: It shouldn't be `true` for all cases where `OutputTransforms` is not nil (though in this case, your test is correct). You should add some more cases to this test function to more thoroughly handle the various branches of the function you're testing.
```
func (qa *QueryAggregators) HasQueryAggergatorBlock() bool {
	return qa != nil && qa.OutputTransforms != nil && (qa.hasLetColumnsRequest() || qa.OutputTransforms.MaxRows > qa.OutputTransforms.RowsAdded)
}
```

Comment (chilah) [reply pkg/segment/structs/segstructs_test.go:70]: Thanks for your comment @AndrewHess . i just resolve it please review again.

**Review by AndrewHess — COMMENTED** [review]
You added some more `assert`s to your one test case, but you should add more different test cases. For example, test when `qa.OutputTransforms.MaxRows > qa.OutputTransforms.RowsAdded` but `qa.hasLetColumnsRequest()` is `false`. Right now you just have one test case.

Also why did you remove your test case for when the QueryAggregators is nil? That seems like a good thing to test

**Comment by chilah** [comment]
> You added some more `assert`s to your one test case, but you should add more different test cases. For example, test when `qa.OutputTransforms.MaxRows > qa.OutputTransforms.RowsAdded` but `qa.hasLetColumnsRequest()` is `false`. Right now you just have one test case.
> 
> Also why did you remove your test case for when the QueryAggregators is nil? That seems like a good thing to test

aw that was my mistake. i brought it back and extract more case. 

Thanks for review.

**Review by AndrewHess — APPROVED** [review]
Looks good
